.github/workflows/intergation-tests.yml

```diff
@@ -0,0 +1,30 @@
+name: "Trigger Integration tests"
+on:
+  push:
+    branches:
+      - master
+      - 'releases/*'
+  pull_request:
+    branches:
+      - master
+      - 'releases/*'
+jobs: 
+    trigger-integration-tests:
+      name: Trigger Integration tests
+      runs-on: ubuntu-latest
+      steps:
+        - name: Extract branch name
+          id: extract_branch
+          run: |
+            echo "##[set-output name=branchname;]$(echo ${GITHUB_REF##*/})"
+
+        - name: Trigger Test run
+          if: github.event.pull_request.base.ref == 'releases/v1' || steps.extract_branch.outputs.branchname == 'releases/v1' || github.event.pull_request.base.ref == 'master' || steps.extract_branch.outputs.branchname == 'master'
```

Comment:
Given that you have kept a script for extra script for fetching the branchname.. you could just move the entire condition logic to the script.
Add something like `##[set-output name=valid_target_branch;](true || false)`

Reply:
The current version must be more readable.